Combine Fido search results from the same client (#8029)
sunpy/sunpy: changelog/8029.feature.rst

> Now responses from the same client are combined a single response for `~sunpy.net.Fido.search`.

Reviewer: ```suggestion
Search results from the same underlying client are combined where possible into a single response when you use `~sunpy.net.Fido.search`.
```

> Adds the ability to combine responses from the same client into a single Table for `~sunpy.net.Fido.search`

Reviewer: ```suggestion
Now responses from the same client are combined a single response for `~sunpy.net.Fido.search`.
```
